Documentation/Usage.md

```diff
@@ -293,19 +294,37 @@ let package = Package(
 Now type `swift build`.
 
 
-### Working on Apps and Packages Side-by-Side
+## Working on Apps and Packages Side-by-Side (Top of the tree development)
 
-If you are developing an app that consumes a package and you need to work on that package simultaneously then you have several options:
+If you are developing an app that consumes a package and you need to work on that package simultaneously then you can use editable packages as a workaround until we have dedicated tooling for it.
 
- 1. Edit the sources that the package manager clones.
+Consider you have a package `foo` which depends on `bar`.
 
-	The sources are cloned visibly into `./Packages` to facilitate this.
+```swift
+import PackageDescription
+
+let package = Package(
+    name: "foo",
+    dependencies: [
+        .Package(url: "http://url/to/bar", majorVersion: 1),
+    ]
+)
+```
+
+If you want to develop on `bar` as well as `foo`, navigate to `foo` directory and put `bar` in edit mode:
+
+    $ cd foo
+    $ swift package edit bar --revision master
+
+Then remove the `bar` directory created in Packages/ and instead create a symbolic link to the path of `bar` package on your filesystem.
```

Comment:
Let's have a comment here on what this will do, if someone has never used edit. Like "This will make sure the package manager uses the sources in your local bar, and does not try and resolve bar based on the version specification in the manifest."